catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/FileSystemStorageProvider.java

```diff
@@ -411,58 +414,80 @@ private ContentItem readContent(URI uri) throws StorageException {
                     "Unable to find file for content ID: " + uri.getSchemeSpecificPart());
         }
 
-        String extension = FilenameUtils.getExtension(file.getFileName()
-                .toString());
+        String filename = file.getFileName()
+                .toString();
+        String extension = FilenameUtils.getExtension(filename);
+        URI reference = null;
         if (REF_EXT.equals(extension)) {
-            extension = FilenameUtils.getExtension(FilenameUtils.removeExtension(file.getFileName()
-                    .toString()));
+            extension = FilenameUtils.getExtension(FilenameUtils.removeExtension(filename));
```

Comment:
Yes, in the case of external references. The real extension, and then a special reference extension is appended to that so we know how to retrieve the real file contents later. I can add a comment.